libraries/classes/User.php

```diff
@@ -133,4 +133,59 @@ public static function isLoggedIn()
       return false;
     }
   }
+
+  public static function isOnline($uid,$token,$address)
+  {
+    $db = new Factory;
+    $mysql = $db->getDBO();
+
+    $res = $mysql->query("SELECT uid FROM user_keys WHERE uid=$uid");
+
+    if($res->num_rows)
+    {
+      $sql = "UPDATE user_keys SET uid=$uid,token='$token',isLoggedin=1 WHERE uid = $uid";
+    }
+    else
+    {
+      $sql = "INSERT INTO user_keys (uid,token,isLoggedin,ip) values ($uid,'$token',1,'$address')";
+    }
+    
+    $result = $mysql->query($sql);
+
+    if($mysql->connect_error)
+    {
+      throw new \Exception("Error Processing Request", $mysql->connect_error);
+      return false;
+    }
+
+    $db->disconnect();
+    return true;
+  }
+
+  public static function goOffline()
+  {
+    $db = new Factory;
+    $mysql = $db->getDBO();
+    $session = new Session;
+    $uid = $session->get('uid');
+
+    $sql = "SELECT token FROM user_keys where uid=$uid and isLoggedin=1";
+    $rslt = $mysql->query($sql);
+    $rslt = $rslt->fetch_assoc();
+
+    if($session->get('token') == $rslt['token'])
+    {
+      $sql = "UPDATE user_keys SET isLoggedin=0 WHERE uid = $uid and isLoggedin=1";
```

Comment:
`UPDATE user_keys SET isLoggedin=0 WHERE uid = $uid and isLoggedin=1 AND ip=$address`